### DISCUSSION TITLE
Enhanced Music Metadata and Lyrics Retrieval for Radio Stations

### DISCUSSION THREAD

**Discussion by UponAnonymous** [SudhanPlayz/Discord-MusicBot#1254] [Ideas]
Currently, when playing a radio station via the bot, there's inconsistency in displaying the station name, sometimes showing it as "undefined." This suggestion aims to enhance this aspect of the bot by enabling it to retrieve metadata for the current radio station, providing information about the currently playing song as well.

In addition, the ability to access lyrics for the current song playing on the radio station and the option to save the song. This functionality could be achieved through either slash commands or buttons integrated below the embed. To illustrate the idea better, I've created a mockup:
![image](https://github.com/SudhanPlayz/Discord-MusicBot/assets/27465814/8f24d799-b0da-40cb-acab-05fd9ac70371)

**Comment by Neko-Life** [comment]
Can you send me a link to the radio site and the link of the radio station you played using the bot?

**Reply by UponAnonymous** [reply]
The link to the radio site that I used is http://www.radiofeeds.co.uk/mp3.asp and the radio station I commonly use is http://icecast.thisisdax.com/CapitalMP3.m3u

**Reply by Neko-Life** [reply]
Interesting, will see how we can implement this. Seems like VLC can fetch the metadata although only the name of the track and station name

**Reply by Neko-Life** [reply]
Also checkout https://github.com/wtfnotavailable/Discord-MusicBot/issues/31 and see if you wanna help and become a contributor

**Reply by UponAnonymous** [reply]
I will, thank you for the offer :)
